File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchWord.java
package net.osmand.search.core;

import net.osmand.data.LatLon;

public class SearchWord {
	private String word;
	private SearchResult result;
	
	public SearchWord(String word, SearchResult res) {
		this.word = word.trim();
		this.result = res;
	}
	
	public ObjectType getType() {
		return result == null ? ObjectType.UNKNOWN_NAME_FILTER : result.objectType;
	}
	
	public String getWord() {
		return word;
	}
	
	public SearchResult getResult() {
		return result;
	}

	public void syncWordWithResult() {
		word = result.wordsSpan != null ? result.wordsSpan : result.localeName.trim();
	}
	
	public LatLon getLocation() {
		return result == null ? null : result.location;
	}
	
	@Override
	public String toString() {
		return word;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/search/core/SearchWord.kt
package net.osmand.shared.search.core

import net.osmand.shared.data.KLatLon

/**
 * A word of a phrase that a search result was selected for.
 *
 * A copy of `SearchWord` in OsmAnd-java, which stays there for android and tools; this copy is for
 * iOS.
 */
class SearchWord(word: String, private val result: SearchResult?) {

	private var word: String = word.trim { it <= ' ' }

	fun getType(): ObjectType? = if (result == null) ObjectType.UNKNOWN_NAME_FILTER else result.objectType

	fun getWord(): String = word

	fun getResult(): SearchResult? = result

	fun syncWordWithResult() {
		word = result!!.wordsSpan ?: result.localeName!!.trim { it <= ' ' }
	}

	fun getLocation(): KLatLon? = result?.location

	override fun toString(): String = word
}
